src/composables/map/location-info.composable.ts

```diff
@@ -0,0 +1,132 @@
+import { watch } from 'vue'
+import { storeToRefs } from 'pinia'
+
+import useMap from '@/composables/map/map.composable'
+import { listen } from 'ol/events'
+import { Coordinate } from 'ol/coordinate'
+import { MapBrowserEvent } from 'ol'
+import { Feature } from 'ol'
+import { Point } from 'ol/geom'
+import VectorLayer from 'ol/layer/Vector'
+import VectorSource from 'ol/source/Vector'
+import StyleStyle from 'ol/style/Style'
+import StyleCircle from 'ol/style/Circle'
+import StyleFill from 'ol/style/Fill'
+import StyleStroke from 'ol/style/Stroke'
+import { useAppStore } from '@/stores/app.store'
+import { useInfoStore } from '@/stores/info.store'
+
+export const DEFAULT_INFO_ZINDEX = 1500
+export const INFO_FEATURE_LAYER_TYPE = 'infoFeatureLayer'
+
+export default function useLocationInfo() {
+  const map = useMap().getOlMap()
+  let holdTimeoutId: number | undefined = undefined
+  let startPixel: Coordinate | null = null
+  const { infoOpen } = storeToRefs(useAppStore())
+  const { locationInfo, hidePointer } = storeToRefs(useInfoStore())
+
+  const infoFeatureLayer = new VectorLayer({
+    source: new VectorSource({
+      features: [] as Feature[],
+    }),
+    // altitudeMode: 'clampToGround',
+    zIndex: DEFAULT_INFO_ZINDEX,
+  })
+  infoFeatureLayer.set('cyLayerType', INFO_FEATURE_LAYER_TYPE)
+  setInfoStyle(infoFeatureLayer)
+  map.addLayer(infoFeatureLayer)
+
+  watch(hidePointer, doHide => {
+    if (doHide) {
+      infoFeatureLayer.getSource()?.clear()
+    } else {
+      if (locationInfo.value) {
+        const feature = new Feature(new Point(locationInfo.value))
+        infoFeatureLayer.getSource()?.addFeature(feature)
+      }
+    }
+  })
+
+  watch(locationInfo, location => {
+    infoFeatureLayer.getSource()?.clear()
+    if (location && !hidePointer.value) {
+      infoOpen.value = true
+      const feature = new Feature(new Point(location))
+      infoFeatureLayer.getSource()?.addFeature(feature)
```

Comment:
This watcher is very similar to the previous one, maybe they can be combined into one.